terraform/modules/gro_ingestion_service/lambdas/src/functions/publish-event.function.ts

```diff
@@ -1,57 +1,72 @@
-import {DynamoDBRecord, DynamoDBStreamEvent, Handler} from "aws-lambda"
-import {mapToEventRecord} from "../models/EventRecord"
-import {request} from "https"
+import { DynamoDBStreamEvent, Handler } from "aws-lambda"
+import { mapToEventRecord } from "../models/EventRecord"
+import { request } from "https"
+import { PublishEvent } from "../models/PublishEvent"
+import { AttributeValue } from "aws-lambda/trigger/dynamodb-stream"
 
 const apiUrl = process.env.API_URL
 const options = {
     hostname: apiUrl,
-    path: "/todos",
+    path: "/events",
     method: "POST",
     port: 443,
     headers: {
         "Content-Type": "application/json",
     },
 }
 
-const publishRecord = async (record: DynamoDBRecord) => {
-    if (record.eventName !== "INSERT" || !record.dynamodb?.NewImage) {
-        return
-    }
-
-    const eventRecord = mapToEventRecord(record.dynamodb.NewImage)
-    const event = {
+const convertToEvent = (image: { [key: string]: AttributeValue }): PublishEvent => {
+    const eventRecord = mapToEventRecord(image)
+    return {
         eventType: "DEATH_NOTIFICATION",
         eventTime: eventRecord.EventTime,
         id: eventRecord.hash
     }
+}
 
-    return new Promise((resolve, reject) => {
-        const req = request(options, res => {
-            res.setEncoding("utf8")
-            let responseBody = ""
-
-            res.on("data", (chunk) => {
-                responseBody += chunk
-            })
-
-            res.on("end", () => {
-                try {
-                    resolve(JSON.parse(responseBody))
-                } catch (err) {
-                    reject(err)
-                }
-            })
-        })
-        req.on("error", err => {
-            reject(err)
+const publishEvent = async (event: PublishEvent) => new Promise((resolve, reject) => {
+    const req = request(options, res => {
+        res.setEncoding("utf8")
+        let responseBody = ""
+
+        res.on("data", (chunk) => {
+            responseBody += chunk
         })
 
-        req.write(event)
-        req.end()
+        res.on("end", () => {
+            try {
+                resolve(JSON.parse(responseBody))
+            } catch (err) {
+                reject({ error: err, event: event })
+            }
+        })
     })
-}
+    req.on("error", err => {
+        reject(err)
+    })
+
+    req.write(event)
+    req.end()
+})
 
 export const handler: Handler = async (event: DynamoDBStreamEvent) => {
-    console.log(`Streaming ${event.Records.filter(r => r.eventName === "INSERT").length} events`)
-    await Promise.all(event.Records.map(publishRecord))
+    const publishEvents = event.Records
+        .filter(r => r.eventName === "INSERT")
```

Comment:
We might be able to make Lambda do these filters  for us https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html#filtering-syntax